### PR TITLE
process host facts for iso runs

### PR DESCRIPTION
**Pull request by chrismeyersfsu**
* Move isolated clean to our final run hook
* ISO and non-iso code path now share the post-fact-processing code

### CODE REVIEW

File: awx/main/tasks.py [hunk 1]
```diff
@@ -995,7 +995,7 @@ def post_run_hook(self, instance, status):
         Hook for any steps to run before job/task is marked as complete.
         '''
 
-    def final_run_hook(self, instance, status, private_data_dir, fact_modification_times):
+    def final_run_hook(self, instance, status, private_data_dir, fact_modification_times, manager_instnace=None):
```
Comment (ryanpetrello) [inline awx/main/tasks.py]: `manager_instnace` typo

File: awx/main/tasks.py [hunk 2]
```diff
@@ -1565,19 +1566,22 @@ def pre_run_hook(self, job):
...
-    def final_run_hook(self, job, status, private_data_dir, fact_modification_times):
+    def final_run_hook(self, job, status, private_data_dir, fact_modification_times, manager_instance=None):
         super(RunJob, self).final_run_hook(job, status, private_data_dir, fact_modification_times)
         if not private_data_dir:
             # If there's no private data dir, that means we didn't get into the
             # actual `run()` call; this _usually_ means something failed in
             # the pre_run_hook method
             return
         if job.use_fact_cache:
+            logger.info("Using fact cache")
```
Comment (ryanpetrello) [inline awx/main/tasks.py]: --

File: awx/main/tasks.py [hunk 3]
```diff
@@ -1088,6 +1088,7 @@ def run(self, pk, **kwargs):
         '''
         self.safe_env = {}
         private_data_dir = None
+        manager_instance = None
```
Comment (ryanpetrello) [inline awx/main/tasks.py]: Can we call this `isolated_manager_instance` everywhere?

File: awx/main/expect/isolated_manager.py [hunk 4]
```diff
@@ -348,5 +348,4 @@ def run(self, instance, private_data_dir, playbook, module, module_args,
             event_data = {'event': 'EOF', 'final_counter': 0}
             event_data.setdefault(self.event_data_key, self.instance.id)
             CallbackQueueDispatcher().dispatch(event_data)
-        self.cleanup()
```
Comment (ryanpetrello) [inline awx/main/expect/isolated_manager.py]: You need to rebase; this moved to `awx.main.isolated.manager`

File: awx/main/tasks.py (hunk 3 above)
Comment (chrismeyersfsu) [reply awx/main/tasks.py]: yep

File: awx/playbooks/check_isolated.yml [hunk 5]
```diff
@@ -18,20 +18,10 @@
...
         mode: pull
+        delete: yes
         recursive: yes
 
     - name: Fail if previous check determined that process is not alive.
       fail:
         msg: "isolated task is still running"
       when: "is_alive.rc == 0"
-
-    - stat: path="{{src}}/facts/"
```
Comment (ryanpetrello) [inline awx/playbooks/check_isolated.yml:28]: I like that we can remove this, good find 👍

File: awx/main/tasks.py [hunk 6]
```diff
@@ -2337,6 +2339,10 @@ def should_use_proot(self, ad_hoc_command):
         '''
         return getattr(settings, 'AWX_PROOT_ENABLED', False)
 
+    def final_run_hook(self, job, status, private_data_dir, fact_modification_times, isolated_manager_instance=None):
+        super(RunJob, self).final_run_hook(job, status, private_data_dir, fact_modification_times)
```
Comment (ryanpetrello) [inline awx/main/tasks.py]: `super(RunAdHocCommand, self)`

**Review by ryanpetrello — APPROVED** [review]

**Pull request merged**